Dockerfile

```diff
@@ -0,0 +1,89 @@
+#   --progress=plain --build-arg NAMESPACE=test --build-arg BUILD_OPTS=--no-default-features .
+
+# To build mainnet
+#   DOCKER_BUILDKIT=1 docker build -t mobilecoin/full-service:0.0.0 \
+#   --progress=plain --build-arg NAMESPACE=prod .
+
+# Build Args:
+# BUILD_OPTS: - '--no-default-features' - Additonal options to cargo build command
+# NAMESPACE: - test|prod - specifies enclave.css files to download.
+# SGX_MODE: - HW|SW - See README
+# IAS_MODE: - PROD|DEV - See README
+# RUSTFLAGS: - '-C target-cpu=penryn' - RUSTFLAGS Environment Variable
+# SIGNED_ENCLAVE_BASE: - enclave-distribution.${NAMESPACE}.mobilecoin.com - base domain for CSS files.
+# SIGSTRUCT_JSON_LOCATION: - production.json - Json file where CSS file paths are located.
+
+# IMPORTANT: Do not add or update OS packages or components in the builder section.
+# In order to create a consistent and verifiable the build environment, only add
+# or update in the mobilecoin/rust-sgx-base image and refer to the image by its hash.
+
+FROM mobilecoin/rust-sgx-base@sha256:5ffeda68b4a3f28370bd2a533c8ada0f95cbc9b8232b8cd5a633b47a19e39cfc as builder
+
+ARG NAMESPACE=test
+ARG SIGNED_ENCLAVE_BASE=enclave-distribution.${NAMESPACE}.mobilecoin.com
+ARG SIGSTRUCT_JSON_LOCATION=production.json
+
+ENV INGEST_ENCLAVE_CSS=/app/ingest-enclave.css
+ENV CONSENSUS_ENCLAVE_CSS=/app/consensus-enclave.css
+
+WORKDIR /app
+
+# Get enclave sigstruct
+RUN  export SIGSTRUCT_JSON=$(curl -sSL https://${SIGNED_ENCLAVE_BASE}/${SIGSTRUCT_JSON_LOCATION}) \
+  && export CONSENSUS_CSS_URL=$(echo ${SIGSTRUCT_JSON} | jq -r .consensus.sigstruct) \
+  && export INGEST_CSS_URL=$(echo ${SIGSTRUCT_JSON} | jq -r .ingest.sigstruct) \
+  && curl https://${SIGNED_ENCLAVE_BASE}/${CONSENSUS_CSS_URL} -o ${CONSENSUS_ENCLAVE_CSS} \
+  && curl https://${SIGNED_ENCLAVE_BASE}/${INGEST_CSS_URL} -o ${INGEST_ENCLAVE_CSS}
+
+COPY . /app/
+
+ARG BUILD_OPTS
+ARG SGX_MODE=HW
+ARG IAS_MODE=PROD
+ARG RUSTFLAGS='-C target-cpu=penryn'
+
+# Build full-service
+RUN  --mount=type=cache,target=/root/.cargo/git \
+     --mount=type=cache,target=/root/.cargo/registry \
+     --mount=type=cache,target=/app/target \
+     cargo
+     cargo build --release -p mc-full-service ${BUILD_OPTS} \
+  && cp /app/target/release/full-service /usr/local/bin/full-service
+
+
+# TODO: Add non-root
+# This is the runtime container.
+# Adding/updating OS will not affect the ability to verify the build environment.
+FROM ubuntu:bionic-20210416
+
+RUN  apt-get update \
+  && apt-get upgrade -y \
+  && apt-get install -y ca-certificates \
+  && apt-get clean \
+  && rm -r /var/lib/apt/lists \
+  && mkdir -p /usr/share/grpc \
+  && ln -s /etc/ssl/certs/ca-certificates.crt /usr/share/grpc/roots.pem
+
+
+COPY --from=builder /usr/local/bin/full-service /usr/local/bin/full-service
+COPY --from=builder /app/*.css /usr/local/bin/
+
+VOLUME /data
+
+EXPOSE 9090
+
+ENV RUST_LOG=info,mc_connection=info,mc_ledger_sync=info
```

Comment:
I suggest this defaults to `info,rustls=warn,hyper=warn,tokio_reactor=warn,mio=warn,want=warn,rusoto_core=error,h2=error,reqwest=error,rocket=error,<unknown>=error`.